WordPress/gutenberg
# QuickEdit: Add slug field control

## What?


Part of https://github.com/WordPress/gutenberg/issues/64519


https://github.com/user-attachments/assets/b488d6b6-ef22-42dd-8bab-2371b728dcd4


This PR adds the `slug` field control to the Page view. Most of the logic is copied from:
- `getPermalinkParts`: https://github.com/WordPress/gutenberg/blob/5682b928356f624f6ed820d066689423d08d5d88/packages/editor/src/store/selectors.js/#L994-L1016
- `isPermalinkEditable`: https://github.com/WordPress/gutenberg/blob/5682b928356f624f6ed820d066689423d08d5d88/packages/editor/src/store/selectors.js/#L937-L944

Currently, these two functions are public selectors and rely on the editor state, so we can't refactor those. I think that for now, we can keep this duplicate code and when the post editor sidebar will be powered by the dataview, we can think to refactor those, but I'm open to feedback on this.

Similar to #64496, this code could be moved to the `@wordpress/core-fields-control` package if we agree to create it.

### Validation

There is an edge-case that I think that we should take care, but I will wait for your feedback before to proceed. When the input is empty, the save button is clickable:

![image](https://github.com/user-attachments/assets/1efb3717-20bb-4189-b786-75cfada30d6d)

This isn’t ideal because a page/post should always have a slug set. Currently, if the input is empty, the original slug is restored on the `onBlur` event, which is not the best UX.


## How?


## Testing Instructions





Ensure that `Quick Edit in DataViews` and `Quick Edit in DataViews` are enabled.

1. Visit the Site Editor.
2. Click on pages.
3. Toggle to the table view.
4. Click on the settings.
5. Click on details panel icon.
6. Ensure that the link field is visible
7. Click on it.
8. Update the slug.
9. Save.
10. Ensure that the page is visible to the new link.
11. Visit `/wp-admin/options-permalink.php`.
12. Select `plain` setting.
13. Save it.
14. Repeat 1-7 steps.
15. Ensure that you can't update the slug.
